packages/cli/src/commands/verify/action.ts

```diff
@@ -0,0 +1,26 @@
+import NetworkController from '../../models/network/NetworkController';
+import { Options, Args } from './spec';
+import ConfigManager from '../../models/config/ConfigManager';
+
+import ProjectFile from '../../models/files/ProjectFile';
+
+export async function action(params: Options & Args & { dontExitProcess: boolean }): Promise<void> {
+  if (process.env.NODE_ENV !== 'test') {
+    const { network } = await ConfigManager.initNetworkConfiguration(params);
+    Object.assign(params, { network });
+  }
+
+  const controller = new NetworkController(params.network, params.txParams, params.networkFile);
+
+  controller.checkLocalContractDeployed(params.contract, true);
+
+  await controller.verifyAndPublishContract(
+    params.contract,
+    params.optimizer,
+    (params as any).optimizerRuns ?? 200,
```

Comment:
Is the `as any` necessary because the property only exists if `optimizer` is true? This is undoing all the benefits of having that sophisticated type, so I think it would be better to use a simpler interface.

Reply:
This is what I meant by not wanting to propagate these changes in the front end all the way to the `Verifier` module. 

The new types nicely reflect the guarantees placed by the new command framework: I'd rather make these initial steps in the right direction and then use them while refactoring the rest of the verify callstack.

Reply:
Okay, sounds good. Ideally we would say `params as OptimizerTrueOptions` or something. Either way though I'm fine with this. We should put in a comment explaining that we expect the rest of verifier to be refactored to accept the fancier types.